index.html

```diff
@@ -1232,14 +1269,44 @@ <h4>Base Proof Serialization (bbs-2023)</h4>
 `nonMandatory` array of strings encoded using the UTF-8 <a>character encoding</a>.
             </li>
             <li>
-Compute the `bbsSignature` using the `Sign` procedure of [[CFRG-BBS-Signature]]
+Compute the `bbsSignature` using the procedures below dependent on the values
+of |commitment_with_proof| and |use_pseudonyms| options.
+              <ol class="algorithm">
+                <li>
+If |commitment_with_proof| is empty and |use_pseudonyms| is false compute the
+`bbsSignature` the `Sign` procedure of [[CFRG-BBS-Signature]]
+with appropriate key material and `bbsHeader` for the `header` and `bbsMessages`
+for the `messages`.
+                </li>
+                <li>
+If |commitment_with_proof| is not empty and |use_pseudonyms| is false compute the
+`bbsSignature` the `Sign` procedure of [[CFRG-Blind-BBS-Signature]]
 with appropriate key material and `bbsHeader` for the `header` and `bbsMessages`
-for the `messages`
+for the `messages`. This provides for "anonymous holder binding".
+                </li>
+                <li>
+If |commitment_with_proof| is empty and |use_pseudonyms| is true. Generate a
+cryptographically random 32 byte |pid| value. Compute the
```

Comment:
I think we should organize this such that you either pass in `pid` or not -- not that we require implementations to generate a random value for it. We can make a suggestion that that's one way to do it, but it should be done externally, IMO, and passed in. So it may be that we *either* have `commitment_with_proof` or `pid` passed here (it's an error to pass both) -- and then just one of them is used internally. This prevents us from being rigidly opinionated on how `pid` is generated.